var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -292,21 +308,59 @@ def bootstrap_args(self):
 
         return args
 
+    def winbootcmake(self, spec):
+        from spack import fetch_strategy, stage
+        urls = {
+            '3': 'https://cmake.org/files/v3.21/cmake-3.21.2-windows-x86_64.zip',
+            '2': 'https://cmake.org/files/v2.8/cmake-2.8.4-win32-x86.zip'
+        }
+        if spec.satisfies('@3.0.2:'):
+            bootstrap_url = urls['3']
+        else:
+            bootstrap_url = urls['2']
+        remote = fetch_strategy.URLFetchStrategy(url=bootstrap_url)
+        bootstrap_stage_path = os.path.join(self.stage.path, "cmake-bootstraper")
+        with stage.Stage(remote, path=bootstrap_stage_path) as bootstrap_stage:
+            remote.stage = bootstrap_stage
+            remote.fetch()
+            remote.expand()
+            shutil.move(bootstrap_stage.source_path, self.stage.source_path)
+
+    def cmake_bootstrap(self):
+        exe_prefix = self.stage.source_path
+        relative_cmake_exe = os.path.join('spack-src', 'bin', 'cmake.exe')
+        return Executable(os.path.join(exe_prefix, relative_cmake_exe))
+
     def bootstrap(self, spec, prefix):
-        bootstrap = Executable('./bootstrap')
-        bootstrap(*self.bootstrap_args())
+        bootstrap_args = self.bootstrap_args()
+        if os.name == 'nt':
```

Comment:
Above we use `sys.platform`  and here we use `os.name` to determine if we're on Windows or not. We should try to be consistent

Reply:
We should decide on on approach and transition all of Spack to use that, we've been fairly inconsistent with this across the Windows feature branch. After a little research, both options will be broken by MingGW and MSYS. I think a check for the substring `NT` in platform.system()` is the only reliable way to handle those platforms. We don't support them at the moment, so maybe that's a non issue. Outside of that, it truly doesn't matter which we use, but I'm personally in favor of `os.name` as it's shorter, but that's purely personal.

Reply:
Would it make sense to create a spack utility function to handle this? (Not necessarily part of this PR). Then we could add special cases or change underlying check and only need to change one location.

Reply:
That's probably the best approach, I'll consolidate this file, and make a different PR for the utility.

Reply:
That sounds good to me